src/methods/zonal.jl

```diff
@@ -108,16 +108,31 @@ function _zonal(f, x::RasterStackOrArray, ::Nothing, data; progress=true, thread
     geoms = _get_geometries(data, geometrycolumn)
     n = length(geoms)
     n == 0 && return []
-    zs = _alloc_zonal(f, x, first(geoms), n; kw...)
-    _run(1:n, threaded, progress, "Applying $f to each geometry...") do i
+    zs = _alloc_zonal(f, x, geoms, n; kw...)
+    start_index = findfirst(x -> !ismissing(x), zs)
+    isnothing(start_index) && return zs
+    _run(start_index:n, threaded, progress, "Applying $f to each geometry...") do i
         zs[i] = _zonal(f, x, geoms[i]; kw...)
     end
     return zs
 end
 
-function _alloc_zonal(f, x, geom, n; kw...)
-    z1 = _zonal(f, x, geom; kw...)
+function _alloc_zonal(f, x, geoms, n; kw...)
+    # Find first non-missing entry
+    n_missing::Int = 0
+    z1 = _zonal(f, x, first(geoms); kw...)
+    for geom in geoms
+        z1 = _zonal(f, x, geom; kw...)
+        if !ismissing(z1)
+            break
+        end
+        n_missing += 1
+    end
     zs = Vector{Union{Missing,typeof(z1)}}(undef, n)
-    zs[1] = z1
+    zs[1:n_missing] .= missing
+    if n_missing == n
+        return zs
+    end
+    zs[n_missing + 1] = z1
```

Comment:
Some comments explaining what this is for would help

Reply:
I've added clarifications, feel free to indicate more locations I can add clarity.